src/hotspot/share/gc/parallel/psParallelCompact.hpp

```diff
@@ -961,6 +961,25 @@ inline void ParMarkBitMapClosure::decrement_words_remaining(size_t words) {
 // regions and regions compacting into themselves.  There is always at least 1
 // region that can be put on the ready list.  The regions are atomically added
 // and removed from the ready list.
+//
+// During compaction, there is a natural task dependency among regions because
+// destination regions may also be source regions themselves.  Consequently, the
+// destination regions are not available for processing until all live objects
+// within them are evacuated to their destinations. These dependencies lead to
```

Comment:
There is a mix of one and two spaces after a full stop (5 times I think) in this and the next paragraph. I recommend using two spaces like in the above paragraphs for uniformity.